Sources/RMBTSpeedGraphView.swift

```diff
@@ -0,0 +1,346 @@
+//  RMBT
+//
+//  Created by Sergey Glushchenko on 10.11.2021.
+//  Copyright © 2021 appscape gmbh. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+@objc final class RMBTSpeedGraphView: UIView {
+    private static let RMBTSpeedGraphViewContentFrame: CGRect = CGRect(x: 34.5, y: 32.5, width: 243.0,  height: 92.0)
+    private static let RMBTSpeedGraphViewSeconds: TimeInterval = 8.0
+    
+    private var backgroundImage: UIImage?
+    
+    private var widthPerSecond: CGFloat = 0.0
+
+    private var backgroundLayer: CALayer = CALayer()
+    private var linesLayer: CAShapeLayer = CAShapeLayer()
+    private var fillLayer: CAShapeLayer = CAShapeLayer()
+    
+    private var graphRect: CGRect {
+        var rect = self.bounds
+        rect.size.width -= 40
+        rect.size.height -= 18
+        return rect
+    }
+    
+    fileprivate var chartPoints: [CGPoint] = []
+    
+    private var value1TopConstraint: NSLayoutConstraint?
+    private lazy var value1Label: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.roboto(size: 11, weight: .regular)
+        label.textColor = labelsColor
+        label.text = "1"
+        return label
+    }()
+    
+    private var value10TopConstraint: NSLayoutConstraint?
+    private lazy var value10Label: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.roboto(size: 11, weight: .regular)
+        label.textColor = labelsColor
+        label.text = "10"
+        return label
+    }()
+    
+    private var value100TopConstraint: NSLayoutConstraint?
+    private lazy var value100Label: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.roboto(size: 11, weight: .regular)
+        label.textColor = labelsColor
+        label.text = "100"
+        return label
+    }()
+    
+    private var value1000TopConstraint: NSLayoutConstraint?
+    private lazy var value1000Label: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.roboto(size: 11, weight: .regular)
+        label.textColor = labelsColor
+        label.text = "1000"
+        return label
+    }()
+    
+    
+    private lazy var startIntervalLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.roboto(size: 11, weight: .regular)
+        label.textColor = labelsColor
+        label.text = "0s"
+        return label
+    }()
+    
+    private lazy var endIntervalLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.roboto(size: 11, weight: .regular)
+        label.textColor = self.labelsColor
+        label.text = "8s"
```

Comment:
It seems that the only difference in those labels above is the `text`. Couldn't we create a subclass for creating the labels taking the `text` as argument, to reduce code repetition? And then that class could have a method to create constraints using the code from the `setup()` method, again, to reduce repetition.

Reply:
@polylina Done